### PR TITLE
Update core.py [If  funnction do not return any thing then warning will be pop up]

### PR DESCRIPTION
**Pull request by saikethan27**
If  `self.fn.process` does not return anything [`None`] then `warning ` will be popped as   
```
_LOGGER.warning(
          no iterator is returned by the process method in DoFn)
```
where changes are committed in function `_check_fn_use_yield_and_return`  to check if yield or return any one is present in code

this Issue is related to #28061

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by saikethan27** [comment]
assign set of reviewers

File: sdks/python/apache_beam/transforms/core.py [hunk 1]
```diff
@@ -1423,15 +1423,21 @@ def _check_fn_use_yield_and_return(fn):
...
+    if len(source_code)==0:
+      return None
     for line in source_code.split("\n"):
       if line.lstrip().startswith("yield ") or line.lstrip().startswith(
           "yield("):
         has_yield = True
       if line.lstrip().startswith("return ") or line.lstrip().startswith(
           "return("):
         has_return = True
+      if line.lstrip().startswith("return None"):
```
Comment (jrmccluskey) [inline sdks/python/apache_beam/transforms/core.py:1435]: The warn could also be placed here instead, eliminating the need for the `elif` block.

File: sdks/python/apache_beam/transforms/core.py [hunk 2]
```diff
@@ -1423,15 +1423,21 @@ def _check_fn_use_yield_and_return(fn):
     source_code = _get_function_body_without_inners(fn)
     has_yield = False
     has_return = False
+    if len(source_code)==0:
+      return None
```
Comment (jrmccluskey) [inline sdks/python/apache_beam/transforms/core.py:1427]: What is the purpose of this check?

**Comment by damccorm** [comment]
I think this is trying to do the same thing as https://github.com/apache/beam/pull/28159 which was opened first and seems slightly more complete, so I think we should proceed with that one and I'm going to close this pull request.

In general, please make sure that you are assigned to an issue before you start working on it to avoid duplicate work. If someone else is assigned but you're not sure if they're working on the problem, its usually best to ask first.

**Pull request closed without merging**